### PR TITLE
Allow user to check off an item that is purchased

### LINKED ISSUE #8: 8. As a user, I want to mark an item on my shopping list as purchased so the app can learn how often I buy different items.
AC:

* User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
* Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

Notes:

* One challenge your pair may work through is figuring out you they will test this.
* E.g., maybe create a few sample items then go into the Firestore console and manipulate the dates. T
* The wirefrrame shows the use of simple checkboxes for this purpose, which is fine for now. https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png

### PR DESCRIPTION
**Pull request by AirZT11**
## Description

* Purchased checkbox is disabled after 24 hours
* Purchased timestamp is deleted from database after 24 hours
* Purchased is set to false in databased after 24 hours

## Related Issue

closes #8 

## Acceptance Criteria
[AC](https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/8)

AC:
* User is able to tap a checkbox or similar UI element to mark an item in the list as purchased
* Item should be shown as checked for 24 hours after the purchase is made (i.e. we assume the user does not need to buy the item again for at least 1 day)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal, pull from main branch with git pull origin main
* Pull this branch with git pull origin sk-dp-check-as-purchased
* Checkout branch with git checkout sk-dp-check-as-purchased

Add an item to the list and then check the checkbox to state that the item is purchased. Go to [firestore](https://console.firebase.google.com/project/tcl-27-shopping-list/firestore/data/~2Fgroceries) and make sure that the submitted item has a purchased boolean value and a checkedTime timestamp value. Edit the checkedTime value to a date/time that is over 24 hours ago and verify that the checkbox in the list is unchecked after refreshing page. Check database one more time and verify that the lastPurchase date has been updated to be the previous checkedTime date/time.

### CODE REVIEW

File: src/components/GroceryCard.js [hunk 1]
```diff
@@ -1,8 +1,65 @@
...
+      });
+      setPurchased(false);
+    } else {
+      ref.update({
+        purchased: true,
+        checkedTime: firebase.firestore.FieldValue.serverTimestamp(),
+      });
+      setPurchased(true);
+    }
+  };
```
Comment (tcelovsky) [inline src/components/GroceryCard.js:33]: This is really cool! Great job figuring out and using FieldValue to update the datestamp!

File: src/components/GroceryCard.js [hunk 2]
```diff
@@ -1,8 +1,65 @@
...
+
+  const purchasedTimeLimit = () => {
+    if (item.checkedTime) {
+      const timeSincePurchased = new Date() - item.checkedTime.toDate();
+      const oneFullDay = 24 * 60 * 60 * 1000;
```
Comment (tcelovsky) [inline src/components/GroceryCard.js:38]: Love the oneFullDay value!

File: src/components/GroceryCard.js [hunk 3]
```diff
@@ -1,8 +1,65 @@
...
+        setPurchased(true);
+      }
+    }
+  };
+
   return (
     <div>
+      <label style={{ display: 'none' }} htmlFor="purchased-checkbox">
+        Purchased
+      </label>
```
Comment (tcelovsky) [inline src/components/GroceryCard.js:55]: I wonder if there's anything we'll need to update here after the lesson on accessibility. I don't know what the rules/guidelines are, so I'll have to look into this for myself.

**Review by tcelovsky — APPROVED** [review]
Great job figuring out how to update/delete datestamp in Firebase! I'd love to learn how FieldValue works, if you have time to talk about that :)

File: src/components/GroceryCard.js (hunk 2 above)
Comment (YolandaHaynes) [reply src/components/GroceryCard.js:38]: I agree with @tcelovsky . I didn't realize you can base the timeframe in milliseconds like that. Something I will def keep in mind in the future.

**Review by YolandaHaynes — APPROVED** [review]
So far everything did what it was supposed to do when I was adjusting the time/date in the firestore database. Good Job!

File: src/components/GroceryCard.js (hunk 2 above)
Comment (aprilleone) [reply src/components/GroceryCard.js:38]: This name could be oneFullDayMS to help know at a glance what you mean by a full day.

**Review by aprilleone — APPROVED** [review]

**Pull request merged**